File: Python_Programming_Assignment_2/Assignment2_4.py
# 4. Write a program which accept one number from user and return additon of its factors.

# Input : 12                    Output : 16         (1+2+3+4+6)

def Add_of_Factors(No):
    sum = 0
    for i in range(1,int(No/2)+1):
        if(No%i == 0):
            print(i, end=" ")
            sum += i
    return sum


def main():
    no = int(input("Enter number to find its factors addition : "))
    print("Addition of Factors of ", no, " is : ", Add_of_Factors(no))

if __name__ == "__main__":
    main()

File: Typescript_Assignment/Typescript_Assignment2/Assignment2_4.ts
// 2.4 Write a typescript program which contains one arrow function named as
//     ChkArmstrong. That function accepts one numbers and check whether 
//     Armstrong number or not.

// Input : 153
// Output : It is Armstrong number.

function ChkArmstrong(num : number) : boolean
{
    var rem : number
    var no : number = num
    var sum : number = 0

    while(no>0)
    {
        rem = no%10
        sum = sum + rem*rem*rem
        no = Math.floor(no/10)
    }

    if(sum == num)
    {
        return true     // if sum is same as given number then return true
    }
    else
    {
        return false    // if sum is not same as given number then return false
    }
}

var num : number = 153

if(ChkArmstrong(num)==true)
{
    console.log(num+" is a Armstrong number")
}
else
{
    console.log(num+" is not a Armstrong number")
}

/*

E:\Typescript\Assignment\Typescript_Assignment2>tsc Assignment2_4.ts

E:\Typescript\Assignment\Typescript_Assignment2>node Assignment2_4.js
153 is a Armstrong number

E:\Typescript\Assignment\Typescript_Assignment2>

*/